Packs/DatadogCloudSIEM/Integrations/DatadogCloudSIEM/DatadogCloudSIEM.py

```diff
@@ -0,0 +1,1361 @@
+    first_fetch_time = params.get("first_fetch", "3 days")
+    fetch_limit = params.get("max_fetch", 50)
+    first_fetch_time = dateparser.parse(f"-{first_fetch_time}")
+    last_run = demisto.getLastRun()
+    with ApiClient(configuration) as api_client:
+        incidents = []
+        api_instance = IncidentsApi(api_client)
+        configuration.unstable_operations["search_incidents"] = True
+
+        response = api_instance.search_incidents(
+            query=incident_serach_query({}),
+            page_size=int(fetch_limit) if int(fetch_limit) < 200 else 200,
+            sort=IncidentSearchSortOrder("-created"),
+        )
+        results = response.to_dict()
+        data = results.get("data", {}).get("attributes", {}).get("incidents", [])
+        data = [convert_datetime_to_str(incident.get("data")) for incident in data]
+        data_list = [
+            incident
+            for incident in data
+            if (
+                datetime.fromisoformat(incident["attributes"]["modified"])
+                .replace(tzinfo=None)
+                .timestamp()
+                > datetime.strptime(
+                    last_run.get("lastRun"), "%Y-%m-%d %H:%M:%S.%f"
+                ).timestamp()
+                if last_run.get("lastRun")
+                else first_fetch_time.timestamp()
+                if first_fetch_time
+                else None
+            )
+        ]
+        for obj in data_list:
+            new_obj = obj["attributes"]
+            new_obj["type"] = obj["type"]
+            new_obj["detected"] = datetime.fromisoformat(
+                obj["attributes"]["detected"]
+            ).strftime(UI_DATE_FORMAT)
+            new_obj["relationships"] = obj["relationships"]
+            new_obj["id"] = obj["id"]
+            new_obj["detection_method"] = obj["attributes"]["fields"][
+                "detection_method"
+            ]["value"]
+            new_obj["root_cause"] = obj["attributes"]["fields"]["root_cause"]["value"]
+            new_obj["summary"] = obj["attributes"]["fields"]["summary"]["value"]
+            new_obj["notification_display_name"] = (
+                obj["attributes"]["notification_handles"][0]["display_name"]
+                if obj["attributes"]["notification_handles"]
+                else None
+            )
+            new_obj["notification_handle"] = (
+                obj["attributes"]["notification_handles"][0]["handle"]
+                if obj["attributes"]["notification_handles"]
+                else None
+            )
+            incident = {
+                "name": obj["attributes"]["title"],
+                "occurred": obj["attributes"]["modified"],
+                "dbotMirrorId": obj["id"],
+                "rawJSON": json.dumps({"incidents": new_obj}),
+                "type": "Datadog Cloud SIEM",
+            }
+            incidents.append(incident)
+    demisto.setLastRun({"lastRun": str(datetime.utcnow())})
```

Comment:
Please in the the last run put the time value of the oldest event that had been fetched